src/code/z_scene.c

```diff
@@ -414,7 +414,7 @@ void Scene_CommandAlternateHeaderList(PlayState* play, SceneCmd* cmd) {
     osSyncPrintf("\n[ZU]sceneset counter=[%X]", ((void)0, gSaveContext.sceneLayer));
 
     if (gSaveContext.sceneLayer != SCENE_LAYER_CHILD_DAY) {
-        altHeader = ((SceneCmd**)SEGMENTED_TO_VIRTUAL(cmd->altHeaders.data))[gSaveContext.sceneLayer - 1];
+        altHeader = ((SceneCmd**)SEGMENTED_TO_VIRTUAL(cmd->altHeaders.data))[gSaveContext.sceneLayer - SCENE_LAYER_CHILD_NIGHT];
```

Comment:
I don't particularly like introducing the enum here without better documenting how this function behaves, because I feel that adding the enum here obfuscates the logic.

Arguably using the enum in this particular spot is incorrect, since it doesn't make sense as to why subtracting `SCENE_LAYER_CHILD_NIGHT` is necessary to access e.g. `SCENE_LAYERP_CUTSCENE_0`, which this section does.

Reply:
It's very clear how this function behaves imo. The child day layer is default, and others are "alternate" starting from child_night (hence the child_night usage here)

I thought about this issue and it could be addressed with `#define SCENE_LAYER_ALTERNATE_HEADER_START SCENE_LAYER_CHILD_NIGHT` though I don't particularly support that solution.

imo `SCENE_LAYER_CHILD_NIGHT` simply maps to the first alternate header (if there are alternate headers) so the subtraction here changes the index from being a layer index to an alternate header index

Reply:
but is there any reason to bind that 1 to `SCENE_LAYER_CHILD_NIGHT`? I'd argue no, considering that the alt header system exists in MM, but the concept of `SCENE_LAYER_CHILD_NIGHT` is completely gone.

Reply:
I have 0 idea how MM works but this is OoT

and I can't answer to why, it's just what the engine does

Reply:
the reason is as dragorn said, its the first of the alternate headers. 
that being said, it may be more worthwhile to define it as the default + 1. This fact will be true no matter how many base "non" cutscene layers there are 

it is true that in mm there is only 1 "gameplay" layer and the rest are cutscenes. In both contexts, the important bit here is that its 1 more than the default layer

Reply:
I think I agree with mzx here and we should probably not use the enum values here. To me this condition and array access is intended as "if not the first layer, get the alt header for it", where the first layer is necessarily 0 and the other layers are >= 1. So I think it makes a lot more sense as `if (layer != 0) get alt header of index (layer - 1)`, and there is no reason to bind child day/night to this, since it should be 0 and 1 regardless